src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,120 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String insertManufacturerRequest = "INSERT INTO manufacturers(name, country) values(?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                     PreparedStatement createManufacturerStatement =
+                             connection.prepareStatement(insertManufacturerRequest,
+                                     Statement.RETURN_GENERATED_KEYS)) {
+            createManufacturerStatement.setString(1, manufacturer.getName());
+            createManufacturerStatement.setString(2, manufacturer.getCountry());
+            createManufacturerStatement.executeUpdate();
+            ResultSet generatedKeys = createManufacturerStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                manufacturer.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert manufacturer "
+                    + manufacturer + " to DB", e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        Manufacturer manufacturer = new Manufacturer();
+        String gettingRequest = "SELECT * FROM manufacturers where manufacturers.id = (?)"
+                + " AND is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getManufacturerStatement =
+                         connection.prepareStatement(gettingRequest)) {
+            getManufacturerStatement.setString(1, String.valueOf(id));
+            ResultSet resultSet = getManufacturerStatement.executeQuery();
+            if (!resultSet.next()) {
+                return Optional.empty();
+            }
+            String name = resultSet.getString("name");
+            String country = resultSet.getString("country");
+            manufacturer.setId(id);
+            manufacturer.setName(name);
+            manufacturer.setCountry(country);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get manufacturer " + id + " from DB ", e);
+        }
+        return Optional.of(manufacturer);
```

Comment:
```suggestion
            ResultSet resultSet = getManufacturerStatement.executeQuery();
            Manufacturer manufacturer = null;
            if (resultSet.next()) {
                //update your manufacturer
            }
            
            return Optional.ofNullable(manufacturer);
        } catch (SQLException e) {
            throw new DataProcessingException("Can't get manufacturer " + id + " from DB ", e);
        }
        
```

you can use Optional of nullable in this section 
somethimg like this